exporter/awsxrayexporter/internal/translator/segment_test.go

```diff
@@ -1075,6 +1075,511 @@ func TestServerSpanWithAwsLocalServiceName(t *testing.T) {
 	assert.False(t, strings.Contains(jsonStr, "user"))
 }
 
+func TestLocalRootConsumer(t *testing.T) {
```

Comment:
Seems like a lot of code duplication in these tests, making it hard for me to review. Can we try to refactor to helper methods to make review easier?

Basically I'm expecting the following tests:
* SERVER LocalRoot - Result: Segment
* PRODUCER LocalRoot - Result: Segment & Remote Subsegment, links on subsegment
* CLIENT LocalRoot - Result: Segment & Remote Subsegment, links on subsegment
* CONSUMER LocalRoot - Result: Segment & Remote Subsegment, links on segment
* INTERNAL LocalRoot - Result: Segment & Remote Subsegment, links on subsegment
* SERVER NonLocalRoot - Result: Segment
* PRODUCER NonLocalRoot with `aws.remote` - Result: Remote Subsegment
* CLIENT NonLocalRoot with `aws.remote` - Result: Remote Subsegment
* CONSUMER:receive NonLocalRoot with `aws.remote`  - Result: Remote Subsegment
* CONSUMER:process NonLocalRoot without `aws.remote`  - Result: Subsegment
* INTERNAL NonLocalRoot with `aws.remote`  - Result: Remote Subsegment

The NonLocalRoot tests should mostly already exist, and just be a matter of checking namespace. Other tests with the same results should have a lot of symmetry and so should be easy to break out common helper logic.

Reply:
"INTERNAL NonLocalRoot with aws.remote - Result: Remote Subsegment"
I believe it should be a not remote subsegment?

Reply:
"INTERNAL LocalRoot - Result: Segment & Remote Subsegment, links on subsegment"
This is in contradiction to the requirements in the SIM. The SIM says the local root internal spans should not have a subsegment.

Please confirm which is correct.

Reply:
You are correct on both accounts, please forgive the copy paste error.